modules/dag-l1/src/main/scala/org/tessellation/dag/l1/http/p2p/L0CurrencyClusterClient.scala

```diff
@@ -0,0 +1,27 @@
+package org.tessellation.dag.l1.http.p2p
+
+import org.tessellation.schema.Block
+import org.tessellation.sdk.http.p2p.PeerResponse
+import org.tessellation.sdk.http.p2p.PeerResponse.PeerResponse
+import org.tessellation.security.signature.Signed
+
+import io.circe.Encoder
+import org.http4s.Method.POST
+import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
+import org.http4s.client.Client
+
+trait L0CurrencyClusterClient[F[_], B <: Block[_]] {
+  def sendL1Output(output: Signed[B]): PeerResponse[F, Boolean]
+}
+
+object L0CurrencyClusterClient {
+
+  def make[F[_], B <: Block[_]: Encoder](pathPrefix: String, client: Client[F]): L0CurrencyClusterClient[F, B] =
```

Comment:
Have you found any refined type for `pathPrefix`?

Reply:
I was thinking about it for a moment but then decided that we will probably figure it out if we decide to add prefixes to all currency endpoints in general. I didn't want to implement partial solution here with a type. We may need multiple things thought through e.g. ADT for prefixes possibly (if limited) or a single refined type or something else, how to inject it in non-boilerplate'y way, should it be in the code or configurable. For now I implemented it similarly to how it's implemented on L0 layer.